pkg/apiclient/argo-server-client.go

```diff
@@ -55,8 +57,12 @@ func (a *argoServerClient) NewInfoServiceClient() (infopkg.InfoServiceClient, er
 	return infopkg.NewInfoServiceClient(a.ClientConn), nil
 }
 
-func NewClientConn(argoServer string) (*grpc.ClientConn, error) {
-	conn, err := grpc.Dial(argoServer, grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(MaxClientGRPCMessageSize)), grpc.WithInsecure())
+func newClientConn(opts ArgoServerOpts) (*grpc.ClientConn, error) {
```

Comment:
I did consider usability, which is why you can set an env var, or you can use `-e`. The `ARGO_SERVER` is `host:port` not  `proto://host:port`, so this would be more complex.

Reply:
I would prefer to make `ARGO_SERVER` smarter to support both `host:port` and `protocol://host:port`. Think about `kubectl`, `curl` and `wget`, they never ask the user to give an argument to indicate that. But it's up to you, I'll approve it.